kernel/src/lib.rs

```diff
@@ -416,3 +424,59 @@ pub trait Engine: AsAny {
     /// Get the connector provided [`ParquetHandler`].
     fn get_parquet_handler(&self) -> Arc<dyn ParquetHandler>;
 }
+
+#[cfg(doctest)]
+mod doc_tests {
```

Comment:
can we add a comment on the requirement to use doctests here?

Reply:
I added a comment, lmk if you had something else in mind